### ISSUE TITLE
netstandard2.0 references projects which is invalid for netframework

### ISSUE THREAD

**Issue by simendsjo** [neo4j/neo4j-dotnet-driver#453]
Assemblies which has been added out-of-band to nuget, should not be referenced for netframework projects. As netstandard2.0 is usef for both netcore and netframework, adding such assemblies to netstandard, will break netframework projects. The versions referenced, will not be the same versions as those included in netframework, so tools will generate wrong binding redirects for instance. These out-of-band packages is a continuous source of problems for netframework projects.

This section is probably what's creating the problems. For netframework, none of these libraries should be referenced.
```xml
  <ItemGroup Condition="'$(TargetFramework)' == 'netstandard2.0'">
    <PackageReference Include="System.Runtime.InteropServices.RuntimeInformation" Version="4.3.0" />
    <PackageReference Include="System.Runtime.Serialization.Primitives" Version="4.3.0" />
    <PackageReference Include="System.Net.NameResolution" Version="4.3.0" />
    <PackageReference Include="System.Net.Security" Version="4.3.2" />
    <PackageReference Include="System.Net.Sockets" Version="4.3.0" />
    <PackageReference Include="System.Threading.Thread" Version="4.3.0" />
  </ItemGroup>
```

Modifying the generated nuspec file manually and maintaining our own package works, but it's better to fix this upstream

```xml
    <dependencies>
      <group targetFramework=".NETStandard2.0">
      </group>
      <group targetFramework=".NETCoreApp2.0">
        <dependency id="System.Net.NameResolution" version="4.3.0" exclude="Build,Analyzers" />
        <dependency id="System.Net.Security" version="4.3.2" exclude="Build,Analyzers" />
        <dependency id="System.Net.Sockets" version="4.3.0" exclude="Build,Analyzers" />
        <dependency id="System.Runtime.InteropServices.RuntimeInformation" version="4.3.0" exclude="Build,Analyzers" />
        <dependency id="System.Runtime.Serialization.Primitives" version="4.3.0" exclude="Build,Analyzers" />
        <dependency id="System.Threading.Thread" version="4.3.0" exclude="Build,Analyzers" />
        <dependency id="System.ValueTuple" version="4.5.0" exclude="Build,Analyzers" />
      </group>
    </dependencies>
```

**Comment by simendsjo** [comment]
FYI, a similar and related discussion is ongoing in Neo4jClient: https://github.com/DotNet4Neo4j/Neo4jClient/issues/383

**Comment by AndyHeap-NeoTech** [comment]
Thanks simensjo. I will take a look and bring myself up to speed.

**Comment by simendsjo** [comment]
I'd like to raise this issue again. Specifying nuget versions of framework assemblies is problematic as versions doesn't line up properly. As an example, you reference "System.Runtime.InteropServices.RuntimeInformation" version 4.3. If you look at the assembly for net45 inside the nupkg, you'll see the assembly has version  4.0.1.0. But when using your library in a netframework project, the framework assembly will (and rightly should) be used instead. This is version 4.0.2.0 for net472. Because you state that 4.0.1.0 should be used in your specification, a redirect for 4.0.1.0 will be created, and thus cannot be used for netframework.

I've yet to find a good solution for this. Seems like Microsoft fixes this in their packages by adding specific netframework sections to avoid referencing packages which exists in netstandard -- which really shouldn't be necessary as netframework should be compatible with netstandard, but because Microsoft f***ed up on the whole netframework/netstandard/netcore migration path, workarounds as these are necessary.

So doing the same as Microsoft is probably the best move. Add a section for netframework without referencing nuget packages which exists in netframework (that is assemblies which is marked as frameworkAssembly in the nuspec along the way -- which is every assembly you use by the looks of it).

I'm also trying to get Paket to add workarounds to avoid generating redirects for assemblies marked as frameworkAssembly for netframework projects as that should hopefully fix packages with this problem.

I would have created a PR for this, but it looks like some of your build is internal..?

**Comment by simendsjo** [comment]
Anything I can do to help? The fix should be trivial by adding a section for netframework and not adding any packages, but I suspect it needs to be done in some of your internal systems?

**Comment by AndyHeap-NeoTech** [comment]
Hi. I will put some time to one side to take a look at this in the next few days. We prioritised some internal feature work that we needed to do for the next release, but I can move this up the list.

**Comment by hkelley** [comment]
@simendsjo ,  do you have a workaround for this?

**Comment by simendsjo** [comment]
> @simendsjo , do you have a workaround for this?

Sorry for the late reply. The workaround is very manual. I download the nupkg (which is just a zip), fix the broken nuspec file (see previous reply), and add the package to an internal nuget repository.

**Comment by thelonelyvulpes** [comment]
Hi I believe this issue was resolved with 4.3.0 release.
https://www.nuget.org/packages/Neo4j.Driver/4.3.0
